_plugins/page_structure.rb

```diff
@@ -0,0 +1,220 @@
+          # try and parse the YAML
+          yamlContent = YAML.load($partitionedFileContent[1])
+          # if we can, use it (section_type needs to be one of the allowed)
+          if yamlContent != nil && yamlContent != false && yamlContent['section_type'] != nil && yamlContent['section_type'] != "default"
+
+            if yamlContent['section_type'] == nil
+              # if no section position has been specified, put it at the end
+              yamlContent['section_position'] = Integer::MAX
+            end
+            # if there's YAML, check it has the section_position tag and put it into child pages
+            ($allPages ||= []) << ChildPage.new(yamlContent, $partitionedFileContent[2..-1].join('---'))
+          end
+        end     
+      end
+      $allPages = sortBySectionPositions($allPages)
+      # return the combined content
+      $allPages
+    end
+          
+    ##
+    # This function looks in a parent folder for all files in the format <Number>_<title>.md
+    # 
+    #
+    def self.parseChildPagesFromParent(page)
+      # get the base directory of the current file
+      $baseFile = Dir.pwd+page['dir']
+      # list all of the files in that directory
+      $listings = Dir[$baseFile+"/*"]
+      # filter by the key pattern
+      $listings = $listings.select{ |i| i[/[\d\.]\_.*\.md/] }
+      # Sort the files based on the Gem::Version of the prefix
+      $listings = $listings.sort{ |x,y| Gem::Version.new((File.basename x).partition('_').first) <=> Gem::Version.new((File.basename y).partition('_').first)  }
+      # loop through them and merge the content
+      $allPages = []
+
+      for $listing in $listings
+        $textContent = ""
+        yamlContent = nil
+                
+        # read the file
+        $fileContent = IO.read($listing)
+        # try and split of any YAML
+        $partitionedFileContent = $fileContent.split('---');
+        # if there's potentially partitioned YAML try and parse it
+        if $partitionedFileContent.size > 2
+          # try and parse the YAML
+          yamlContent = YAML.load($partitionedFileContent[1])
+          # if we can, use it
+          if yamlContent != nil && yamlContent != false
+            $textContent = $partitionedFileContent[2..-1].join('---')
+          end
+        end
+        
+        # if there's no text content set yet, just use the whole file
+        if $textContent == ""
+          # use the whole file content
+          $textContent = $fileContent
+        end
+        # append the current file to the content
+        ($allPages ||= []) << ChildPage.new(yamlContent, $textContent)
+                
+      end
+      # return the combined content
+      $allPages
+    end
+  end
+  
+  class IncludePageContentTag < Liquid::Tag
+      def initialize(tag_name, text, tokens)
+        super
+        @text = text.strip
+      end
+      def render(context)
+#        $childPages = ChildPage.parseChildPagesFromParent(context['page'])
```

Comment:
Why commented out? Either include a TODO to explain why line is useful (and when it would be uncommented), or delete it.